Updated version to 3.5 everywhere and created a script to use in the future

deepakaravindr opened the pull request:
1. Updated version from 3.4.0 to 3.5.0 on the vsix manifest version
   and NuGetPackage.cs
2. Simplified common.props file to have the
   SemanticVersion as a single property.
3. Created an UpdateVersion script that can be used to update NuGet version 
   when creating new releases.

@yishaigalatzer @emgarten @alpaix

alpaix commented on scripts/utils/UpdateNuGetVersion.ps1:
```diff
@@ -0,0 +1,41 @@
+ls -r project.json | ?{ ReplaceNuGetVersion($_.FullName) }
+
+$vsixManifestFile = Join-Path $nuGetRoot ".\src\NuGet.Clients\VsExtension\source.extension.vsixmanifest"
+ReplaceNuGetVersion($vsixManifestFile)
```
`.\` is redundant.

deepakaravindr replied:
Done

alpaix commented on scripts/utils/UpdateNuGetVersion.ps1:
```diff
@@ -0,0 +1,41 @@
+}
+
+Write-Host "Changing directory to $nuGetRoot.."
+cd $nuGetRoot
```
Is this needed? Script passes full paths everywhere.

deepakaravindr replied:
True. Removed it

alpaix commented on build/common.props:
```diff
@@ -31,9 +31,7 @@
-		<SemanticVersionMajor>3</SemanticVersionMajor>
-		<SemanticVersionMinor>5</SemanticVersionMinor>
-		<SemanticVersionPatch>0</SemanticVersionPatch>
+		<SemanticVersion>3.5.0</SemanticVersion>
```
:+1:

emgarten commented:
:shipit:

deepakaravindr committed:
1. Simplified common.props file to have the
SemanticVersion as a single property.
2. Created an UpdateVersion script that can be used
when creating new releases.
3. Updated version from 3.4.0 to 3.5.0 on the vsix manifest version
and NuGetPackage.cs

With the new script though, it will be an easy and safe change.

alpaix commented:
:shipit:

deepakaravindr commented:
Merged with https://github.com/NuGet/NuGet.Client/commit/3e7620feae5f9ca7c3410daf10b6701ab71f40de
